draft-ietf-jose-hpke-encrypt.md

```diff
@@ -183,19 +183,76 @@ Encapsulated keys MUST be the base64url encoded encapsulated key as defined in S
 
 In HPKE JWE Integrated Encryption, JWE Encrypted Key is the encapsulated key.
 
-In HPKE JWE Key Encryption, each recipient JWE Encrypted Key is the encrypted content encryption key, and the encapsulated key is found in the recipient header.
+In HPKE JWE Key Encryption, each recipient JWE Encrypted Key is the encrypted content encryption key, and the encapsulated key (ek) is found in the recipient header.
 
 # Integrated Encryption
 
 In HPKE JWE Integrated Encryption:
 
 - The protected header MUST contain an "alg" that starts with "HPKE".
-- The protected header MUST contain an "enc" that is registered in both the IANA HPKE AEAD Identifiers Registry, and the IANA JSON Web Signature and Encryption Algorithms Registry.
+- The protected header MUST contain an "enc" and it must be set to the value "dir".
 - The protected header parameters "psk_id" and "auth_kid" MAY be present.
 - The protected header parameters "ek" MUST NOT be present.
 - The "encrypted_key" MUST be the base64url encoded encapsulated key as defined in Section 5.1.1 of {{RFC9180}}.
 - The "iv", "tag" and "aad" members MUST NOT be present.
 - The "ciphertext" MUST be the base64url encoded ciphertext as defined in Section 5.2 of {{RFC9180}}.
+- The HPKE Setup info parameter MAY be used and its values are not constrained by this specification, by default it is empty.
```

Comment:
we will have to remove the following line from the draft: 
The "Setup info" MUST NOT be used with either HPKE JWE Integrated Encryption and HPKE JWE Key Encryption.

Reply:
I think for interoperability we should say the following for both modes :
When apu / apv are present, the HPKE Setup info parameter will carry the JOSE context-specific data defined in Section 4.6.2 of [RFC7518].